diff --git a/scripts/cmake/DependenciesExternalProject.cmake b/scripts/cmake/DependenciesExternalProject.cmake
index 0e1470fe249..87961bcd3b6 100644
--- a/scripts/cmake/DependenciesExternalProject.cmake
+++ b/scripts/cmake/DependenciesExternalProject.cmake
@@ -63,7 +63,9 @@ if(OGS_USE_MFRONT)
         set(_py_boost_comp
             "python${Python_VERSION_MAJOR}${Python_VERSION_MINOR}"
         )
-        find_package(Boost ${ogs.minimum_version.boost} COMPONENTS ${_py_boost_comp})
+        find_package(
+            Boost ${ogs.minimum_version.boost} COMPONENTS ${_py_boost_comp}
+        )
         if(Boost_${_py_boost_comp}_FOUND)
             set(_tfel_cmake_args
                 "-DPython_ADDITIONAL_VERSIONS=${_py_version_major_minor}"
@@ -270,13 +272,14 @@ if(NOT ZLIB_FOUND)
             ${build_dir_ZLIB}/${CMAKE_INSTALL_BINDIR}/zlib${_zlib_debug_postfix}${CMAKE_SHARED_LIBRARY_SUFFIX}
         )
         # requires CMake 3.24 to be effective:
-        set(ZLIB_USE_STATIC_LIBS "ON")
-        set(ZLIB_ROOT ${build_dir_ZLIB})
+        set(ZLIB_USE_STATIC_LIBS "ON" CACHE BOOL "" FORCE)
+        set(ZLIB_ROOT ${build_dir_ZLIB} CACHE PATH "" FORCE)
         # Force local zlib build, found netcdf-installed zlib sometimes
         set(ZLIB_LIBRARIES
             ${build_dir_ZLIB}/${CMAKE_INSTALL_LIBDIR}/zlibstatic${_zlib_debug_postfix}${CMAKE_STATIC_LIBRARY_SUFFIX}
+            CACHE STRING "" FORCE
         )
-        set(ZLIB_INCLUDE_DIRS ${build_dir_ZLIB})
+        set(ZLIB_INCLUDE_DIRS ${build_dir_ZLIB} CACHE STRING "" FORCE)
     endif()
     set(_EXT_LIBS ${_EXT_LIBS} ZLIB CACHE INTERNAL "")
 endif()
